---
title: "Exploratory analysis clean from DB"
output: html_document
date: "2023-02-21"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(RSQLite)
library(tidyverse)
library(caret)
library(magrittr)
library(glue)
library(cowplot)
library(modelr)
```
## Reading in the data
Read in the data, and set the data types to the correct types
Drop index number from the pandas dataframe and NCBI.tax.ID
```{r}
# Connect to the DB
conn <- dbConnect(SQLite(),"../../../s16.sqlite")
# List of all the tables
dbListTables(conn)

D_tmp <- dbGetQuery(conn, "SELECT * FROM ribdif_bacdive_joined")
D_tmp <- tibble(D_tmp, .name_repair ="universal")

D_tmp <- mutate(D_tmp, across(antibiotics:PH.range, factor))
D_tmp <- select(D_tmp, !c(NCBI.tax.ID,strain.designation))
```



# Splitting up data

```{r}
set.seed(25022023)
# Adding ID as a column
D_tmp %<>% mutate(ID = row_number(species))
# Randomly selecting the training/exploration data with seed set
D <- D_tmp %>% slice_sample(prop = 0.7)
# Assigning the rest of the data to the test dataset
D_test <- anti_join(D_tmp, D, by = "ID")

# Checking if its correctly split up
percent_in_test <- nrow(D_test)/(nrow(D)+nrow(D_test))
percent_in_test

```


## Looking at data distribution
### For different taxonomic scales
TODO: also make plot of just n species

```{r}
# Function to plot how often each tax is seen
taxplot <- function(D, tax, name,ylimit){
  plot <- D %>% 
    group_by({{tax}}) %>%
    summarise(n=n()) %>% 
    mutate(tax_ordered = fct_reorder({{tax}},desc(n))) %>% 
    ggplot() +
    geom_segment(aes(x=tax_ordered, xend = tax_ordered,y=0, yend=n)) +
    xlab(glue("taxonomic {name}")) + ylab("Number of entries") +
    theme(axis.text.x=element_blank(),axis.ticks.x=element_blank()) +
    ggtitle(glue("Entries pr. {name}")) +
    ylim(0,ylimit)
  return(plot)
}

# Plloting the different taxs
p1 <- taxplot(D, family, "family",350)
p2 <- taxplot(D, order, "order",350)
p3 <- taxplot(D, class, "class",2000)
p4 <- taxplot(D, phylum, "phylum",2000)
p5 <- taxplot(D, genus, "genus",350)
plot_grid(p5,p1, p2, p3, p4,labels ="auto")
#gridExtra::grid.arrange(p5,p1,p2,p3,p4)
```
The goal of this part is to figure out how to take into consideration that some strains are sequences a lot more than others.
I have already calculated the mean/mode of each value at the species level, but it seems that we still have an unequal distribution.
Furthermore it seems that nomatter how high the taxonomic scale we go, There is still an unequal distribution. Even though wee took into 
consideration that some species are sequences more we would still except an unequeal distribution. Based on this and the fact that information is lost
for each "step" on the taxonomic scale I have choosen to go with the speceis level
It is relevant as when we for example look at the distribution of numbers of 16s genes, more sequences genes will have a higher weight.

```{r}
taxboxplot <- function(D, tax, n16_var, name){
  D_new <- D %>% 
    group_by({{tax}}) %>%
    summarise(mean_value=mean({{n16_var}}))
  plot <- ggplot(D_new) +
    geom_histogram(aes(x=mean_value, y=after_stat(density))) +
    ggtitle(glue("density pr. {name}"))+
    xlab("mean n16") + ylab("density") +
    stat_function(fun = dnorm, col = "red", args = list(mean = mean(D_new$mean_value), sd = sd(D_new$mean_value)))
  return(plot)
}


# Plloting the different taxs
to_plot <- "n16"
p0 <- taxboxplot(D, species, .data[[to_plot]],"species")
p1 <- taxboxplot(D, genus, .data[[to_plot]],"genus")
p2 <- taxboxplot(D, family,.data[[to_plot]], "family")
p3 <- taxboxplot(D, order, .data[[to_plot]],"order")
p4 <- taxboxplot(D, class,.data[[to_plot]], "class")
p5 <- taxboxplot(D, phylum,.data[[to_plot]], "phylum")

plot_grid(p0,p1, p2, p3, p4,p5,labels ="auto")


```
But the most important thing is the distribution. Here we can see that for species it looks a bit wierd, it looks better for genus
Lets have a look at the log transformed

```{r}
# Not working:: Error in if (!is.finite(log_num)) { : the condition has length > 1
L2log <- function(num){
  log_num = log2(num)
  print(log_num)
  if (!is.finite(log_num)){return(0)}
  else{
  return (log_num)}
}

functaxboxplot <- function(D, tax, n16_var, name,func){
  D_new <- D %>% 
    group_by({{tax}}) %>%
    summarise(mean_value=mean({{n16_var}}))
  plot <- ggplot(D_new) +
    geom_histogram(aes(x=func(mean_value), y=after_stat(density)),bins = 13) +
    ggtitle(glue("density pr. {name}"))+
    xlab("logn16") + ylab("density") +
    stat_function(fun = dnorm, col = "red", args = list(mean = mean(func(D_new$mean_value)), sd = sd(func(D_new$mean_value))))
  return(plot)
}


# Plloting the different taxs
to_plot <- "n16"
p0 <- functaxboxplot(D, (species), .data[[to_plot]],"species",log)
p1 <- functaxboxplot(D, (genus), .data[[to_plot]],"genus",log)
p2 <- functaxboxplot(D, (family),.data[[to_plot]], "family",log)
p3 <- functaxboxplot(D, (order), .data[[to_plot]],"order",log)
p4 <- functaxboxplot(D, (class),.data[[to_plot]], "class",log)
p5 <- functaxboxplot(D, (phylum),.data[[to_plot]], "phylum",log)

plot_grid(p0,p1, p2, p3, p4,p5,labels ="auto")


```
Here the distributions look ok for species. And again to not lose information im going with species.s

#### D
Lets have a look at the distributions for D

```{r}

taxboxplot <- function(D, tax, n16_var, name){
  D_new <- D %>% 
    group_by({{tax}}) %>%
    summarise(mean_value=mean({{n16_var}}))
  plot <- ggplot(D_new) +
    geom_histogram(aes(x=mean_value, y=after_stat(density))) +
    ggtitle(glue("Entries pr. {name}"))+
    xlab("div") + ylab("density") +
    stat_function(fun = dnorm, col = "red", args = list(mean = mean(D_new$mean_value), sd = sd(D_new$mean_value)))
  return(plot)
}

to_plot <- "div"
p0 <- taxboxplot(D, species, .data[[to_plot]],"species")
p1 <- taxboxplot(D, genus, .data[[to_plot]],"genus")
p2 <- taxboxplot(D, family,.data[[to_plot]], "family")
p3 <- taxboxplot(D, order, .data[[to_plot]],"order")
p4 <- taxboxplot(D, class,.data[[to_plot]], "class")
p5 <- taxboxplot(D, phylum,.data[[to_plot]], "phylum")

plot_grid(p0,p1, p2, p3, p4,p5,labels ="auto")


```
They do not look normal distributed.
Lets try to log transform it

```{r}


functaxboxplot <- function(D, tax, n16_var, name,func){
  D_new <- D %>% 
    group_by({{tax}}) %>%
    summarise(mean_value=mean({{n16_var}}))
  plot <- ggplot(D_new) +
    geom_histogram(aes(x=func(mean_value), y=after_stat(density)),bins = 20) +
    ggtitle(glue("density pr. {name}"))+
    xlab("logdiv") + ylab("density") +
    stat_function(fun = dnorm, col = "red", args = list(mean = mean(func(D_new$mean_value)), sd = sd(func(D_new$mean_value))))
  return(plot)
}

# Plloting the different taxs
to_plot <- "div"
p0 <- functaxboxplot(D, (species), .data[[to_plot]],"species",log1p)
p1 <- functaxboxplot(D, (genus), .data[[to_plot]],"genus",log1p)
p2 <- functaxboxplot(D, (family),.data[[to_plot]], "family",log1p)
p3 <- functaxboxplot(D, (order), .data[[to_plot]],"order",log1p)
p4 <- functaxboxplot(D, (class),.data[[to_plot]], "class",log1p)
p5 <- functaxboxplot(D, (phylum),.data[[to_plot]], "phylum",log1p)

plot_grid(p0,p1, p2, p3, p4,p5,labels ="auto")

```

Here we can see we have alot of divs at = 0,
Looking at not having them we get this
```{r}

# Plloting the different taxs
to_plot <- "div"
p0 <- functaxboxplot(D, (species), .data[[to_plot]],"species",log)
p1 <- functaxboxplot(D, (genus), .data[[to_plot]],"genus",log)
p2 <- functaxboxplot(D, (family),.data[[to_plot]], "family",log)
p3 <- functaxboxplot(D, (order), .data[[to_plot]],"order",log)
p4 <- functaxboxplot(D, (class),.data[[to_plot]], "class",log)
p5 <- functaxboxplot(D, (phylum),.data[[to_plot]], "phylum",log)

plot_grid(p0,p1, p2, p3, p4,p5,labels ="auto")
```
There is an argument for not working with the var of 0 or we can try and remove all entries with n16=1 (since they always has zero)
It turns out n16=1 gives a ton with no var, n16 > 4 is more acurate. Alternativly you could have it for one speciefic n16 or take n16 into the model when modelling div. But genereally it would be stupid to remove all of this information
```{r}
# For n16 > 4
D_sub <- filter(D, n16 > 4)
to_plot <- "div"
p0 <- functaxboxplot(D_sub, (species), .data[[to_plot]],"species",log1p)
p1 <- functaxboxplot(D_sub, (genus), .data[[to_plot]],"genus",log1p)
p2 <- functaxboxplot(D_sub, (family),.data[[to_plot]], "family",log1p)
p3 <- functaxboxplot(D_sub, (order), .data[[to_plot]],"order",log1p)
p4 <- functaxboxplot(D_sub, (class),.data[[to_plot]], "class",log1p)
p5 <- functaxboxplot(D_sub, (phylum),.data[[to_plot]], "phylum",log1p)
plot_grid(p0,p1, p2, p3, p4,p5,labels ="auto")

# For n16 = 4
D_sub <- filter(D, n16 == 10)
to_plot <- "div"
p0 <- functaxboxplot(D_sub, (species), .data[[to_plot]],"species",log1p)
p1 <- functaxboxplot(D_sub, (genus), .data[[to_plot]],"genus",log1p)
p2 <- functaxboxplot(D_sub, (family),.data[[to_plot]], "family",log1p)
p3 <- functaxboxplot(D_sub, (order), .data[[to_plot]],"order",log1p)
p4 <- functaxboxplot(D_sub, (class),.data[[to_plot]], "class",log1p)
p5 <- functaxboxplot(D_sub, (phylum),.data[[to_plot]], "phylum",log1p)

plot_grid(p0,p1, p2, p3, p4,p5,labels ="auto")

```

```{r}

# Lets try for all different n16. To see how the distribution look for each

D_sub <- mutate(D, n16 = factor(as.integer(n16)))
ggplot(D_sub) +
    geom_histogram(aes(x=log1p(div), y=after_stat(density))) +
  facet_wrap(~n16)


```
1 n16 always has zero, ofc. It seems the more n16 we get the less with zero div. Which again makes sense
Furthermore i believe im going to continue working from a speceis level as the distributions look ok at that level and it is not worth sacreficing resolution to get better model assumptions.. for now

## Simple relationship
We expect there to be a relationship between div and the number of 16s rDNA. 
Since an increase in 16s rDNA means that there are more copies to have a difference between them.
Div is calculated as the 

### VIGTIGT :https://stats.stackexchange.com/questions/278953/linear-regression-on-exponential-distributed-dependent-variable
## OG library(Amelia)
#missmap(training.data.raw, main = "Missing values vs observed")
# https://www.r-bloggers.com/2015/09/how-to-perform-a-logistic-regression-in-r/

```{r}
select(D, species, genus, div, n16) %T>% summary() %>% filter(div > 100)
# Div has a very large unrealistic max And some are very large eg. div = 128 for n16 =# for Tenuifilum thalassicum


par(mfrow=c(2,2))
plot(lm(log1p(div) ~ (n16), D))

par(mfrow=c(2,2))
plot(lm(log1p(div) ~ sqrt(n16), D))


plot(log(D$div), log(D$n16))


Dscuf <- D %>% 
  mutate(Tn16=log(n16), Tdiv=log(div)) 
par(mfrow=c(2,2))
plot(lm(log(Tdiv) ~ log(Tn16), Dscuf))

```


## Modeling Div
The goal of this part is to build a model which takes into consideration 
n16 and taxonomic relationships as it seems they might have a big impact.
We could either:
1) Remove all positions with n16 = 0, as they are not going to include any information about 
the relationship between bacterial ecology and div. Here the intercept could now be set to (0,0) or not
2) Fit a model with a varying intercept
3) Fit a model with 0,0 as intercept as described in the next paragraph
Below we can see div against n16.
The main takeway is that we have a lot small values and a few large, therefore there is an arguemt
for applying a transformation to both axis.

```{r}
ggplot(D,aes(x=n16, y=div)) + 
  geom_point(size = 0.2) +
  geom_smooth()
```
## Model with varying intercept

```{r}
Dt <- D %>% 
  mutate(Tn16=log(n16), Tdiv=log1p(div)) 

# Looking at the distribution of the plot
ggplot(Dt,aes(x=n16, y=Tdiv)) + 
  geom_point(size=0.2) +
  geom_smooth(method=lm, formula = "y~x") 

```



## Model which go through 0
We know that the true intercept should have a div at 0 when n16 is 1
Log transforming n16 means that when n16 = 1 then log(1) becomes 0
div: transform div with log1p means that 0 div becomes log(0+1)=0
Lets have a look at a plot of the transformed variables

```{r}
Dt <- D %>% 
  mutate(Tn16=log(n16), Tdiv=log1p(div)) 

# Looking at the distribution of the plot
ggplot(Dt,aes(x=Tn16, y=Tdiv)) + 
  geom_point(size=0.2) +
  geom_smooth()
```
### Taxonomic information in the model
And for each phylum with over 20 entries
Here it seems that there could be some gain in including phylum in the model, as it seems to have an effect

```{r}
# Plotting for different phylum
Dt %>% 
  group_by(phylum) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 50) %>% 
  ggplot(aes(x=Tn16, y=Tdiv)) +
    geom_point(size=0.5, alpha=0.6) +
    theme(legend.position="none") +
    facet_wrap(~phylum) +
    geom_smooth(method=lm, formula = "y~x+0") +
    ggtitle("By phylum")


```
Lets also have a look for class, 
```{r}
library(magrittr)
# Plotting for different orders
Dt %>% 
  group_by(class) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 20) %>% 
  ggplot(aes(x=Tn16, y=Tdiv)) +
    geom_point(size=0.5, alpha=0.6) +
    theme(legend.position="none") +
    facet_wrap(~class) +
    geom_smooth(method=lm, formula = "y~x+0") +
    ggtitle("By class")

```
Here it's hard to see how much information we lose by just including phylum instead of order. Therefore lets
try and remove the effect of phylum by plotting the residuals of a simple model

```{r}
fitTaxPhylum <- lm(Tdiv ~ 0 + Tn16 + Tn16:factor(phylum)  ,Dt)
summary(fitTaxPhylum)
res <- Dt %>% 
  add_residuals(fitTaxPhylum)

res %>% 
  group_by(class) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 20) %>% 
  ggplot(aes(x=Tn16, y=resid)) +
    geom_point(size=0.5, alpha=0.6) +
    theme(legend.position="none") +
    facet_wrap(~class) +
    geom_smooth(method=lm, formula = "y~x+0") +
    geom_ref_line(h=0, col = "red", size = 0.1) +
    ggtitle("Residuals vs n16 By class")


```
The residuals seems ok distributed. It seems to make sense stay at the phylum level just based on this
But checking the amount of entries in both it seems that they are about the same.
So this effect could be due to each phylum just having one class.
```{r}
print("class:")
res %>% 
  group_by(class) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 20) %>% 
  nrow()
print("phylum:")
res %>% 
  group_by(phylum) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 20) %>% 
  nrow()

```
Lets check
```{r}
res %>% 
  group_by(phylum) %>% 
  reframe(uClass = unique(class)) %>% 
  ggplot() + geom_bar(aes(x=phylum)) +coord_flip()


```
This seems to not be the case
We could also do the same for order.
Here there is more varibilty. But it will argue that we get closer to just predicting the datapoints directly instead of 
the tendency. Therefore i am going to just keep the model at the phylum

```{r}
fitTaxPhylum <- lm(Tdiv ~ 0 + Tn16 + Tn16:phylum ,Dt)
summary(fitTaxPhylum)
res <- Dt %>% 
  add_residuals(fitTaxPhylum)

res %>% 
  group_by(order) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 20) %>% 
  ggplot(aes(x=Tn16, y=resid)) +
    geom_point(size=0.5, alpha=0.6) +
    theme(legend.position="none") +
    facet_wrap(~order) +
    geom_smooth(method=lm, formula = "y~x+0") +
    geom_ref_line(h=0, col = "red", size = 0.1) +
    ggtitle("Residuals vs n16 By order")
```







We can see the points in (0,0)
Lets try to fit a basic model, just of the two, and with phylum

```{r}
D
```



```{r}
Dt <- Dt %>% 
  group_by(phylum) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 20)
# Lets add it to the model
Dt
fitTaxPhylum <- lm(Tdiv ~ 0 + Tn16 + Tn16*Tn16 + Tn16^3 + Tn16:phylum ,Dt)
summary(fitTaxPhylum)
res <- Dt %>% 
  add_residuals(fitTaxPhylum)

res %>% 
  ggplot(aes(resid)) + geom_histogram(bins = 40,binwidth = 0.1) 

res %>% 
  ggplot(aes(x=exp(Tn16) ,y=resid)) +
  geom_hex(bins=50) +
  geom_ref_line(h=0, col = "red", size = 1) +
  geom_smooth()

par(mfrow=c(2,2))
plot(fitTaxPhylum )

# We have on with a very large amount of n16
#Dsub %>% filter(n16>20)

#I checked and its also high here
#https://www.arb-silva.de/search/
#Tumebacillus avium
# we observe more var at the start in the res since they are predicting wrong

```
We can observe that we tend to overestimate the div on genera with larger amount of #16s. 
And we tend to underestimate div for genera with samller amounts of #16s.


QUICK check with dif model

```{r}
Dt <- Dt %>% 
  filter(n16 > 1)
# Lets add it to the model
Dt
fitTaxPhylum <- lm(Tdiv ~ Tn16 + Tn16:phylum ,Dt)
summary(fitTaxPhylum)
res <- Dt %>% 
  add_residuals(fitTaxPhylum)

res %>% 
  ggplot(aes(resid)) + geom_histogram(bins = 40,binwidth = 0.1) 

res %>% 
  ggplot(aes(x=exp(Tn16) ,y=resid)) +
  geom_hex(bins=50) +
  geom_ref_line(h=0, col = "red", size = 1) +
  geom_smooth()

par(mfrow=c(2,2))
plot(fitTaxPhylum )

# We have on with a very large amount of n16
#Dsub %>% filter(n16>20)

#I checked and its also high here
#https://www.arb-silva.de/search/
#Tumebacillus avium
# we observe more var at the start in the res since they are predicting wrong
```

```{r}
Dt <- Dt %>% 
  filter(n16 > 1)
# Lets add it to the model
Dt
fitTaxPhylum <- lm(Tdiv ~ Tn16 ,Dt)
summary(fitTaxPhylum)
res <- Dt %>% 
  add_residuals(fitTaxPhylum)

res %>% 
  ggplot(aes(resid)) + geom_histogram(bins = 40,binwidth = 0.1) 

res %>% 
  ggplot(aes(x=exp(Tn16) ,y=resid)) +
  geom_hex(bins=50) +
  geom_ref_line(h=0, col = "red", size = 1) +
  geom_smooth()

par(mfrow=c(2,2))
plot(fitTaxPhylum )

# We have on with a very large amount of n16
#Dsub %>% filter(n16>20)

#I checked and its also high here
#https://www.arb-silva.de/search/
#Tumebacillus avium
# we observe more var at the start in the res since they are predicting wrong
```
```




### SAMPLING
Lets try and get an idea about the effect of where it is samples from
```{r}

Dt <- D %>% 
  mutate(Tn16=log(n16), Tdiv=log1p(div)) 
Dt <- Dt %>% 
  group_by(phylum) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 20)

Denv <- Dt %>% mutate(aquaP = aquatic.counts/Total.samples , 
              animalP = animal.counts/Total.samples, 
              plantP = plant.counts/Total.samples, 
              soilP = soil.counts/Total.samples)

fitTaxPhylum <- lm(Tdiv ~ Tn16 + Tn16:phylum -1 ,Denv)
res <- Denv %>% 
  add_residuals(fitTaxPhylum)
summary(fitTaxPhylum)


p1 <- ggplot(res, aes(x=aquaP, y=resid)) +
  geom_point()
p2 <- ggplot(res, aes(x=plantP, y=resid)) +
  geom_point()
p3 <- ggplot(res, aes(x=animalP, y=resid)) +
  geom_point()
p4 <- ggplot(res, aes(x=soilP, y=resid)) +
  geom_point()
plot_grid(p1, p2, p3, p4,labels ="auto")

```
It seems that there is no difference here

## Antibiotics, motility, PH, gramstain motility

```{r}

Dt <- D %>% 
  mutate(Tn16=log(n16), Tdiv=log1p(div)) 

Dt <- Dt %>% 
  group_by(phylum) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 20)

Dt <- mutate(Dt, antibiotics = ifelse(is.na(antibiotics), "PNR", antibiotics))

fitTaxPhylum <- lm(Tdiv ~ Tn16 + Tn16:phylum -1 ,Dt)
res <- Dt %>% 
  add_residuals(fitTaxPhylum)
summary(fitTaxPhylum)



p1 <- ggplot(res, aes(x=gram.stain, y=resid)) +
  geom_violin(draw_quantiles = c(0.25, 0.5, 0.75))
p2 <- ggplot(res, aes(x=PH.range, y=resid)) +
  geom_violin(draw_quantiles = c(0.25, 0.5, 0.75)) +
  geom_point(size=0.2)
p3 <- ggplot(res, aes(x=motility, y=resid)) +
  geom_violin(draw_quantiles = c(0.25, 0.5, 0.75)) +
  geom_point(size=0.2)
p4 <- ggplot(res, aes(x=antibiotics, y=resid)) +
  geom_violin(draw_quantiles = c(0.25, 0.5, 0.75)) +
  geom_point(size=0.2)
plot_grid(p1, p2, p3, p4,labels ="auto")
```


```{r}

ggplot(res, aes(x=cefoxitin, y=resid)) +
  geom_boxplot(draw_quantiles = c(0.25, 0.5, 0.75)) +
  geom_point(size=0.2)

```



```{r}
colnames(Dt)
# Lets try with
fitTaxPhylum <- lm(Tdiv ~ 0 + Tn16 + Tn16:phylum + oxygen.tolerance:Tn16 + Tn16:oxygen.tolerance, Dt)
res <- Dt %>% 
  add_residuals(fitTaxPhylum)

res %>% 
  ggplot(aes(resid)) + geom_histogram(bins = 40,binwidth = 0.1) 

res %>% 
  ggplot(aes(x=exp(Tn16) ,y=resid)) +
  geom_hex(bins=50) +
  geom_ref_line(h=0, col = "red", size = 1) +
  geom_smooth()
```








### Div v n16
We expect there to be a relationship

## Antibiotics
Lets start by checking in general
The hypthesis was that they might have more div. 

```{r}
p1 <- ggplot(D)+
  geom_boxplot(aes(x=antibiotics, y=log1p(div)))
p2 <- ggplot(D)+
  geom_boxplot(aes(x=antibiotics, y=log(div)))
p3 <- ggplot(D)+
  geom_boxplot(aes(x=antibiotics, y=log(n16)))
p4 <- ggplot(D)+
  geom_boxplot(aes(x=antibiotics, y=(n16)))
plot_grid(p1,p2,p3,p4,labels="auto")
```
There seems to be now inherint difference

### Formatting daTA
Lets check for more specifc types of AR
First getting the subset of the data with AR resistence info about the Antibiotics
which target the 16s rRNA
```{r}
# Getting the ones which are actually targeting 16S 
# Reading them from ARtarget16s.csv
target16S <- read_csv2("../data/ARtarget16s.csv",show_col_types = FALSE,col_names = FALSE)
targetvector <- as.array(target16S$X1)
found_16S <- as.array(colnames(select(D,lincomycin:spiramycin.II)))
intersect <- intersect(targetvector,found_16S)
D_ar <- select(D_tmp, all_of(intersect), n16, div)

```
### Different types
#### Div
Now lets look at some plots
firstly for div
```{r}
library(gridExtra)

plotlist = list()
for(i in seq_along(intersect)){
  antibiotic = intersect[i]
  p <- ggplot(D_ar)+
    geom_boxplot(aes(x=.data[[antibiotic]], y=log1p(div)))
  plotlist = c(plotlist, list(p))
}

wrap_plots(plotlist)
```
#### n16
and now for n16
```{r}
plotlist = list()
for(i in seq_along(intersect)){
  antibiotic = intersect[i]
  p <- ggplot(D_ar)+
    geom_boxplot(aes(x=.data[[antibiotic]], y=log(n16)))
  plotlist = c(plotlist, list(p))
}

wrap_plots(plotlist)
```
### Deep dive into topramycin
Some seem to have a change but it is my suspesion that its because of the randomness with a small subset randomly having more or just because of other factors. Eg the small subset which are Antibiotic resistenet to a specific AB might just share taxonmic. Lets check for tobramycin
```{r}
# Get the taxonomy info for all the different bacteria resistent to tobramycin
D_ar_tax <- select(D_tmp, intersect, n16, div,genus,family,order, class,phylum)
filter(D_ar_tax, tobramycin == "R") %>% 
  select(genus,family,order,class,phylum) 
# It seems most of them share the Gammaproteobacteria class / are from the..


GorNot <- mutate(D_ar_tax, isGamma = ifelse(class == "Gammaproteobacteria","Gamma","notGamma")) %>% 
  mutate(isGamma=factor(isGamma)) 

p1 <- ggplot(GorNot) +
geom_boxplot(aes(x=isGamma, y=log1p(div)))

p2 <- ggplot(D_ar)+
    geom_boxplot(aes(x=tobramycin, y=log1p(div)))
                 

p3 <- ggplot(GorNot) +
geom_boxplot(aes(x=isGamma, y=log(n16)))

p4 <- ggplot(D_ar)+
    geom_boxplot(aes(x=tobramycin, y=log(n16)))
          
plot_grid(p1,p2,p3,p4)


```

So we can see that it's proberly just tax- information messing it up
Statistically we should test types of AR agains tax.
Here we have 895 different in Gammaproteo group but only 8 of them with the AR

```{r}
#Lastly lets try and plot the ones with AR against the ones which do not for gammaproteo
# Its mostly saying for the ones which are Gammaproteo

GorNot %>% 
  ggplot()+
    geom_boxplot(aes(x=tobramycin, y=log1p(div))) +
    facet_wrap(~isGamma)

# There actually seem to be more
# Lest see for n16
# We expect n16 to go up for R again since the are correlated
GorNot %>% 
  ggplot()+
    geom_boxplot(aes(x=tobramycin, y=(n16))) +
    facet_wrap(~isGamma)
# It seems to actually go down ! 
# so there is something going on :I

```

Lets try and remove the tax (maybe phylum (could be class)) information

```{r}
library(modelr)
D_ar_tax %>% 
  ggplot() +
  geom_boxplot(aes(x=phylum, y=log1p(div)))

D_ar_tax %<>% mutate(phylum = factor(phylum)) 
# Fit the model
tax_model <- lm(log1p(div) ~phylum ,data=D_ar_tax)
summary(tax_model)

# Visualize the predictions
pred <- D_ar_tax %>% 
  data_grid(phylum) %>% 
  add_predictions(tax_model)
ggplot() +
  geom_boxplot(aes(y=phylum, x=log1p(div)), data=D_ar_tax) +
  geom_point(aes(y=phylum, x=pred), data=pred, col="red")

# Visulize the residuals
res <- D_ar_tax %>% 
  add_residuals(tax_model)
# no need to specify x,y in aes()
# We can see the effect here
res %>% 
  ggplot(aes(y=resid, x=tobramycin)) +
  geom_boxplot()

# Looking at the res there is still a lot of uneqplanied especially under
res %>% 
  ggplot(aes(resid)) +
  geom_histogram()
D_tmp
# Lets see where it comes from
res %>% 
  ggplot(aes(n16 ,resid)) +
  geom_point(size=0.5) +
  geom_ref_line(h=0, col = "red", size = 1)



```

```{r}
library(magrittr)
# trying to take into consideration the n16
tax_n16_model <- update(tax_model, ~.+n16)
# Now it looks alot better:
D_ar_tax %>% 
  add_residuals(tax_n16_model) %>% 
  ggplot(aes(resid)) + geom_histogram(bins = 40,binwidth = 0.1) 

res <- D_ar_tax %>% 
  add_residuals(tax_n16_model) 
  
plotlist = list()
for(i in seq_along(intersect)){
  antibiotic = intersect[i]
  p <- ggplot(res)+
    geom_boxplot(aes(x=.data[[antibiotic]], y=resid))
  plotlist = c(plotlist, list(p))
}

wrap_plots(plotlist)


```

## Lets see for class
```{r}

# making the lm model
tax_n16_model <- lm(log1p(div) ~ class+n16, D_ar_tax)
# This looks better than before. Maybe go all the way down?
D_ar_tax %>% 
  add_residuals(tax_n16_model) %>% 
  ggplot(aes(resid)) + geom_histogram(bins = 40,binwidth = 0.1) 

res <- D_ar_tax %>% 
  add_residuals(tax_n16_model) 
  
plotlist = list()
for(i in seq_along(intersect)){
  antibiotic = intersect[i]
  p <- ggplot(res)+
    geom_boxplot(aes(x=.data[[antibiotic]], y=resid))
  plotlist = c(plotlist, list(p))
}

wrap_plots(plotlist)


res %>% 
  ggplot(aes(log(n16) ,resid)) +
  geom_point(size=0.5) +
  geom_ref_line(h=0, col = "red", size = 1) +
  geom_smooth()

```

# Lets try with the log(n16)
```{r}

D_ar_sub <- mutate(D_ar_tax,logdiv = log1p(div), logn16 = log(n16))

# making the lm model
tax_n16_model <- lm(logdiv ~ phylum + logn16, data=D_ar_sub)

# This looks better than before. Maybe go all the way down taxonomically?
D_ar_sub %>% 
  add_residuals(tax_n16_model) %>% 
  ggplot(aes(resid)) + geom_histogram(bins = 40,binwidth = 0.1) 

res <- D_ar_sub %>% 
  add_residuals(tax_n16_model) 
  

#plotlist = list()
#for(i in seq_along(intersect)){
#  antibiotic = intersect[i]
#  p <- ggplot(res)+
#    geom_boxplot(aes(x=.data[[antibiotic]], y=resid))
#  plotlist = c(plotlist, list(p))
#}
#
#wrap_plots(plotlist)


res
res %>% 
  ggplot(aes(2^logn16 ,resid)) +
  geom_hex(bins=50) +
  geom_ref_line(h=0, col = "red", size = 1) +
  geom_smooth()
# Here it looks good, might be because some genera only have like two species in them..

```

FOR 
```{r}
D_ar_sub <- mutate(D_ar_tax,logdiv = log1p(div), logn16 = log(n16))

# making the lm model
tax_n16_model <- lm(logdiv ~ order + logn16, data=D_ar_sub)

# This looks better than before. Maybe go all the way down taxonomically?
D_ar_sub %>% 
  add_residuals(tax_n16_model) %>% 
  ggplot(aes(resid)) + geom_histogram(bins = 40,binwidth = 0.1) 

res <- D_ar_sub %>% 
  add_residuals(tax_n16_model) 
  

#plotlist = list()
#for(i in seq_along(intersect)){
#  antibiotic = intersect[i]
#  p <- ggplot(res)+
#    geom_boxplot(aes(x=.data[[antibiotic]], y=resid))
#  plotlist = c(plotlist, list(p))
#}
#
#wrap_plots(plotlist)


res
res %>% 
  ggplot(aes(2^logn16 ,resid)) +
  geom_hex(bins=50) +
  geom_ref_line(h=0, col = "red", size = 1) +
  geom_smooth()
# Here it looks good, might be because some ORDERs only have like two species in them..


# Lets look at order distribution
D_ar_tax %>% 
  group_by(phylum) %>% 
  summarise(n = n()) %>% 
  mutate(tax_ordered = fct_reorder(phylum,desc(n))) %>% 
  filter(n < 10) %>% 
  ggplot() +
  geom_segment(aes(x=tax_ordered, xend = tax_ordered,y=0, yend=n))

s

# 10 out of 34 tax have under 10 entries. Largest Pseudomonadota have 1800
D_ar_tax %>% 
  group_by(phylum) %>% 
  summarise(n = n()) %>% 
  mutate(tax_ordered = fct_reorder(phylum,desc(n))) %>% 
  filter(n < 10) %>% 
  nrow()


D_ar_tax %>% 
  group_by(phylum) %>% 
  summarise(n = n()) %>% 
  mutate(tax_ordered = fct_reorder(phylum,desc(n))) %>% 
  nrow()


D_ar_tax %>% 
  group_by(phylum) %>% 
  summarise(n = n()) %>% 
  mutate(tax_ordered = fct_reorder(phylum,desc(n))) %>% 
  arrange(desc(n))

nrow(D_tmp)
```
Probelemet ift tax er at det for de sm√• tax basically bare giver dem svaret



## Larger dataset
```{r}
dbListTables(conn)

Dl_tmp <- dbGetQuery(conn, "SELECT * FROM species_gcf2species_ribdif_info")
Dtax <- dbGetQuery(conn, "SELECT * FROM taxInfoFull") %>% rename(genus = GENUS, phylum=PHYLUM, class=CLASS ,family=FAMILY)

Dl <- left_join(Dl_tmp, Dtax, by="genus")
```


```{r}
colnames(Dl)
ggplot(Dl,aes(x=mean)) +
  geom_histogram(bins = 100)

```



## Genome size v total seq length

```{r}

Dlt <- Dl %>% 
  mutate(Tn16=log(number_16s), Tdiv=log1p(total_div))

## for phylum
Dlt %>% 
  group_by(phylum) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 100) %>% 
  ggplot(aes(x=Tn16, y=Tdiv)) +
    geom_hex(size=0.5, alpha=0.6) +
    theme(legend.position="none") +
    geom_smooth(method=lm, formula = "y~x+0") +
    facet_wrap(~phylum) 

## for class
Dlt %>% 
  group_by(class) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 100) %>% 
  ggplot(aes(x=Tn16, y=Tdiv)) +
    geom_hex(size=0.5, alpha=0.6) +
    theme(legend.position="none") +
    geom_smooth(method=lm, formula = "y~x+0") +
    facet_wrap(~class) 

```












